### PR TITLE
MODINVSTOR-364: holdingsRecord.permanentLocationId is required field

### PR DESCRIPTION
**Pull request by julianladisch**

### CODE REVIEW

**Comment by marcjohnson-kint** [comment]
This change will require pull requests for the following modules to update compatibility:
folio_inventory-1.13.1000608
folio_plugin-create-item-1.3.100016
mod-circulation-17.0.0-SNAPSHOT.383
mod-inventory-13.1.0-SNAPSHOT.173
mod-orders-8.1.0-SNAPSHOT.225
mod-patron-3.0.3-SNAPSHOT.45
mod-rtac-1.3.0-SNAPSHOT.38

**Comment by marcjohnson-kint** [comment]
Also, I think if we are going to make a breaking change, we should also the RAML to include the validation trait and apply the UUID regular expressions for the relevant properties (part of https://issues.folio.org/browse/MODINVSTOR-297)

**Comment by bohdan-suprun** [comment]
@marcjohnson-kint, I've added UUID patterns in holdingsrecord.json.

**Comment by bohdan-suprun** [comment]
@marcjohnson-kint, I've created PRs to update API version in Java modules, so only  UIIN-774 and UIPCITEM-2 left.

**Comment by marcjohnson-kint** [comment]
Do all of the methods that implement endpoints in `holdings-storage` have the `@Validate` trait applied?

**Comment by bohdan-suprun** [comment]
> Do all of the methods that implement endpoints in `holdings-storage` have the `@Validate` trait applied?

@marcjohnson-kint, they applied to the generated interface, do we need to apply them to the class that implements the methods?

**Comment by marcjohnson-kint** [comment]
> > Do all of the methods that implement endpoints in `holdings-storage` have the `@Validate` trait applied?
> 
> @marcjohnson-kint, they applied to the generated interface, do we need to apply them to the class that implements the methods?

I don't know for certain. @skoczko @adamdickmeiss @julianladisch Can you confirm whether it is sufficient for the interface to be attributed, or do the implementation methods also need it?

**Comment by mreno-EBSCO** [comment]
> > > Do all of the methods that implement endpoints in `holdings-storage` have the `@Validate` trait applied?
> > 
> > 
> > @marcjohnson-kint, they applied to the generated interface, do we need to apply them to the class that implements the methods?
> 
> I don't know for certain. @skoczko @adamdickmeiss @julianladisch Can you confirm whether it is sufficient for the interface to be attributed, or do the implementation methods also need it?

The implementation methods need the `@Validate` annotation.

See bullet 1: https://github.com/folio-org/raml-module-builder#implement-the-interfaces

**Comment by bohdan-suprun** [comment]
Thanks @mreno-EBSCO! @marcjohnson-kint, I've added `@Validate` annotation to holdings-storage implementation.

File: src/test/java/org/folio/rest/api/HoldingsStorageTest.java [hunk 1]
```diff
@@ -369,6 +374,72 @@ public void canGetAllHoldings()
...
+
+    Response response = getCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody().trim(),
+      is("'offset' parameter is incorrect. parameter value {-3} is not valid: must be greater than or equal to 0"));
+  }
+
+  @Test
+  public void cannotDeleteHoldingWhenLangParameterIsInvalid() throws Exception {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: What makes `eng` an invalid value for language?

**Comment by marcjohnson-kint** [comment]
I couldn't spot a test for trying to create a holdings record without a permanent location. If there isn't one, please could one be added.

File: src/test/java/org/folio/rest/api/HoldingsStorageTest.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: it have to be a two symbol i.e. `en`

**Comment by bohdan-suprun** [comment]
> I couldn't spot a test for trying to create a holdings record without a permanent location. If there isn't one, please could one be added.

@marcjohnson-kint, I've added.

File: src/test/java/org/folio/rest/api/HoldingsStorageTest.java (hunk 1 above)
Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: Thanks

Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/HoldingsStorageTest.java]:
```suggestion
  public void cannotDeleteHoldingWhenLangParameterIsTooLong() throws Exception {
```

**Pull request merged**